cni.go

```diff
@@ -154,16 +154,39 @@ func (c *libcni) Setup(ctx context.Context, id string, path string, opts ...Name
 	return c.createResult(result)
 }
 
+type asynchAttachResult struct {
+	res *types100.Result
+	err error
+}
+
+func asynchAttach(ctx context.Context, n *Network, ns *Namespace, wg *sync.WaitGroup, rc chan asynchAttachResult) {
+	defer wg.Done()
+	r, err := n.Attach(ctx, ns)
+	rc <- asynchAttachResult{res: r, err: err}
+}
+
 func (c *libcni) attachNetworks(ctx context.Context, ns *Namespace) ([]*types100.Result, error) {
+	var wg sync.WaitGroup
+	var lastError error
 	var results []*types100.Result
+	rc := make(chan asynchAttachResult)
+
 	for _, network := range c.Networks() {
-		r, err := network.Attach(ctx, ns)
-		if err != nil {
-			return nil, err
+		wg.Add(1)
+		go asynchAttach(ctx, network, ns, &wg, rc)
+	}
+
+	for range c.Networks() {
+		rs := <-rc
+		if rs.err != nil {
+			lastError = rs.err
+		} else {
+			results = append(results, rs.res)
```

Comment:
Is ordering here important or at all relevant? `results` could be initialized with the final expected length and keep the index if so.

Reply:
> Is ordering here important or at all relevant? `results` could be initialized with the final expected length and keep the index if so.

good point been programming in go to long crutching on append  :-)